util/testutil/integration/frombinary.go

```diff
@@ -0,0 +1,57 @@
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/content/local"
+	"github.com/containerd/containerd/images/archive"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+func providerFromBinary(fn string) (_ ocispec.Descriptor, _ content.Provider, _ func(), err error) {
+	ctx := context.TODO()
+
+	tmpDir, err := ioutil.TempDir("", "buildkit-state")
+	if err != nil {
+		return ocispec.Descriptor{}, nil, nil, err
+	}
+	close := func() {
+		os.RemoveAll(tmpDir)
+	}
+	defer func() {
+		if err != nil {
+			close()
+		}
+	}()
+	// can't use contentutil.Buffer because ImportIndex has wrong dependency signature
```

Comment:
could you add more context about "wrong dependency signature"

Reply:
It takes a `content.Store` even though only `Ingester/Provider` is required. `Buffer` does not implement `Store`.